source/adios2/toolkit/sst/dp/rdma_dp.c

```diff
@@ -798,6 +1027,11 @@ static int get_cxi_auth_key_from_env(CP_Services Svcs, void *CP_Stream, struct _
     char const *slingshot_devices = getenv("SLINGSHOT_DEVICES");
     char const *preferred_device = get_preferred_domain(Params);
 
+    if ((!preferred_device && strncmp("cxi", preferred_device, 3) != 0) || !slingshot_devices)
```

Comment:
```suggestion
    if ((!preferred_device || strncmp("cxi", preferred_device, 3) != 0) || !slingshot_devices)
```
I need to check this logic again, currently this is wrong

Reply:
Should be fixed with ba7d6c1